inductiva/resources/machine_groups.py

```diff
@@ -133,3 +132,34 @@ def get():
         machine_group_list.append(mg_class.from_api_response(mg))
 
     return machine_group_list
+
+
+def get_machine_type(num_cpus: int,
+                     ram_gb: Optional[int] = None,
+                     spot: bool = False):
+    """Get the machine type with the lowest price.
+
+    If the machine type with the exact requirements is not found, the closest
+    machine type with higher resources will be returned. For example, if a
+    machine type with 5 CPUs and 20 GB of RAM is requested, but the closest
+    machine type has 8 CPUs and 32 GB of RAM, the latter will be returned.
+
+    Args:
+        num_cpus: The minimum number of CPUs required.
+        ram_gb: The minimum RAM in GB required. If None, the minimum RAM for
+          the given number of CPUs will be used.
+        spot: Whether the machine is spot or not.
+    """
+    spot = "t" if spot else "f"
+    api = compute_api.ComputeApi(inductiva.api.get_client())
+    body = {"num_cpus": num_cpus, "spot": spot}
+    body["ram_gb"] = ram_gb if ram_gb else None
```

Comment:
But if not passed, is None automatically. I think this doesn't matter and then we should just have:

```python
    body = {"num_cpus": num_cpus, "ram_gb": ram_gb, "spot": spot}
```